File: articles/self-control-commitment-devices-behavioral-economics/fortran/commitment_choice.f90
program commitment_choice
  implicit none
  real :: beta, delta, future_benefit, immediate_temptation, commitment_cost
  real :: patient_value, temptation_value
  integer :: periods, patient_choice

  beta = 0.72
  delta = 0.97
  future_benefit = 1000.0
  immediate_temptation = 600.0
  commitment_cost = 300.0
  periods = 12

  patient_value = beta * (delta ** periods) * future_benefit
  temptation_value = immediate_temptation - commitment_cost

  if (patient_value >= temptation_value) then
    patient_choice = 1
  else
    patient_choice = 0
  end if

  print *, "Synthetic patient choice under commitment:", patient_choice
end program commitment_choice

File: articles/self-control-commitment-devices-behavioral-economics/rust/commitment_choice.rs
fn discounted_future_value(beta: f64, delta: f64, benefit: f64, periods: i32) -> f64 {
    beta * delta.powi(periods) * benefit
}

fn choose_commitment(beta: f64, delta: f64, future_benefit: f64, immediate_temptation: f64, commitment_cost: f64, periods: i32) -> bool {
    let patient_value = discounted_future_value(beta, delta, future_benefit, periods);
    let temptation_value = immediate_temptation - commitment_cost;
    patient_value >= temptation_value
}

fn main() {
    println!(
        "Synthetic patient choice under commitment: {}",
        choose_commitment(0.72, 0.97, 1000.0, 600.0, 300.0, 12)
    );
}
